src/common/io.ts

```diff
@@ -1420,6 +1422,20 @@ export class ARIOWriteable extends ARIOReadable implements AoARIOWrite {
     params: AoBuyRecordParams,
     options?: WriteOptions,
   ): Promise<AoMessageResult> {
+    // spawn a new ANT if not provided
+    if (params.processId === undefined) {
+      try {
+        params.processId = await ANT.spawn({
+          signer: this.signer,
+          ao: this.process.ao,
+          logger: this.logger,
+        });
+      } catch (error) {
+        throw new Error('Failed to spawn ANT for name purchase.', error);
+      }
+    }
+
```

Comment:
@kunstmusik mentioned an emitter or callback here would be helpful, but I think for those workflows users can still just manually spawn, and then handle the control flow outside of this.

That being said an `onSigningProgress` call would be easy here, along with in the spawnANT function for verification.

Thoughts?